components/Templates/includes/auto-template/Pods_Templates_Auto_Template_Front_End.php

```diff
@@ -0,0 +1,368 @@
+<?php
+if ( class_exists( 'Pods_PFAT_Frontend' ) ) {
+	return;
+}
+
+/**
+ * Class Pods_Templates_Auto_Template_Front_End
+ *
+ * Replaces Pods_PFAT_Frontend
+ */
+class Pods_Templates_Auto_Template_Front_End {
+	function __construct() {
+		/**
+		 * Allows plugin to append/replace the_excerpt
+		 *
+		 * Default is false, set to true to enable.
+		 */
+		if ( !defined( 'PFAT_USE_ON_EXCERPT' ) ) {
+			define( 'PFAT_USE_ON_EXCERPT', false );
+		}
+
+		if( !is_admin() ){
+			add_action( 'wp', array( $this, 'set_frontier_style_script' ) );
+		}
+
+		add_filter( 'the_content', array( $this, 'front' ), 10.5 );
+
+		if (  PFAT_USE_ON_EXCERPT  ) {
+			add_filter( 'the_excerpt', array ( $this, 'front' ) );
+		}
+
+	}
+
+	/**
+	 * Get all post type and taxonomy Pods
+	 *
+	 * @since 2.4.5
+	 *
+	 * @return array Of Pod names.
+	 */
+	function the_pods() {
+
+		//use the cached results
+		$key = 'pods_pfat_the_pods';
+		$the_pods = pods_transient_get( $key  );
+
+		//check if we already have the results cached & use it if we can.
+		if ( false === $the_pods ) {
+			//get all post type pods
+			$the_pods = pods_api()->load_pods( array(
+					'type' => array(
+						'taxonomy',
+						'post_type'
+					),
+					'names' => true )
+			);
+
+			//cache the results
+			pods_transient_set( $key, $the_pods );
+
+		}
+
+		return $the_pods;
+
+	}
+
+	/**
+	 * Get all Pods with auto template enable and its settings
+	 *
+	 * @return array With info about auto template settings per post type
+	 *
+	 * @since 2.4.5
+	 */
+	function auto_pods() {
+		/**
+		 * Filter to override all settings for which templates are used.
+		 *
+		 * Note: If this filter does not return null, all back-end settings are ignored. To add to settings with a filter, use 'pods_pfat_auto_pods';
+		 *
+		 * @param array $auto_pods Array of parameters to use instead of those from settings.
+		 *
+		 * @return array Settings arrays for each post type.
+		 *
+		 * @since 2.4.5
+		 */
+		$auto_pods = apply_filters( 'pods_pfat_auto_pods_override', null );
+		if ( !is_null( $auto_pods ) ) {
+			return $auto_pods;
+		}
+
+		//try to get cached results of this method
+		$key = 'pods_pfat_auto_pods';
+		$auto_pods = pods_transient_get( $key );
+
+		//check if we already have the results cached & use it if we can.
+		if ( $auto_pods === false  ) {
+			//get possible pods
+			$the_pods = $this->the_pods();
+
+			//start output array empty
+			$auto_pods = array();
+
+			//loop through each to see if auto templates is enabled
+			foreach ( $the_pods as $the_pod => $the_pod_label ) {
+				$pods = pods_api( $the_pod );
```

Comment:
@josh This should be set above the foreach as `$api = pods_api();` and inside the foreach on this line it should have `$pod_data = $api->load_pod( array( 'name' => $the_pod ) );`

Further below instead of `$pods->pod_data` you would just use `$pod_data`

Maybe @pglewis can make this tweak for you today so we can get the release out today.